src/components/Executions.js

```diff
@@ -0,0 +1,215 @@
+    t: PropTypes.func.isRequired,
+    dispatchGetDatasets: PropTypes.func.isRequired,
+    dispatchGetAlgorithms: PropTypes.func.isRequired,
+    dispatchExecuteAlgorithm: PropTypes.func.isRequired,
+    isLoading: PropTypes.bool.isRequired,
+    classes: PropTypes.shape({
+      formControl: PropTypes.string.isRequired,
+      content: PropTypes.string.isRequired,
+      selectLabel: PropTypes.string.isRequired,
+      infoAlert: PropTypes.string.isRequired,
+    }).isRequired,
+  };
+
+  static defaultProps = {
+    datasets: null,
+    algorithms: null,
+  };
+
+  state = (() => {
+    const { datasets, algorithms } = this.props;
+    return {
+      datasetId: datasets?.first()?.id,
+      algorithmId: algorithms?.first()?.id,
+    };
+  })();
+
+  componentDidMount() {
+    const { dispatchGetDatasets, dispatchGetAlgorithms } = this.props;
+    dispatchGetDatasets();
+    dispatchGetAlgorithms();
+  }
+
+  componentDidUpdate({ datasets: prevDatasets, algorithms: prevAlgorithms }) {
+    const { datasets, algorithms } = this.props;
+    if (prevDatasets !== datasets && datasets.size > 0) {
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({ datasetId: datasets.first().id });
+    }
+    if (prevAlgorithms !== algorithms && algorithms.size > 0) {
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({ algorithmId: algorithms.first().id });
+    }
+  }
+
+  handleDatasetSelectOnChange = (e) => {
+    this.setState({ datasetId: e.target.value });
+  };
+
+  handleAlgorithmSelectOnChange = (e) => {
+    this.setState({ algorithmId: e.target.value });
+  };
+
+  executeAlgorithm = () => {
+    const { dispatchExecuteAlgorithm, algorithms } = this.props;
+    const { datasetId, algorithmId } = this.state;
+    const { language } = algorithms.find(({ id }) => id === algorithmId);
+    if (language) {
+      dispatchExecuteAlgorithm({ datasetId, algorithmId, language });
+    }
+  };
+
+  render() {
+    const { datasetId, algorithmId } = this.state;
+    const { datasets, algorithms, t, classes, isLoading } = this.props;
+
+    if (isLoading) {
+      return (
+        <Main fullScreen>
+          <Loader />
+        </Main>
+      );
+    }
+
+    if (datasets.size <= 0) {
+      return (
+        <Main>
+          <Alert severity="info" className={classes.infoAlert}>
+            {t('Load a dataset first')}
+          </Alert>
+        </Main>
+      );
+    }
+
+    if (algorithms.size <= 0) {
+      return (
+        <Main>
+          <Alert severity="info" className={classes.infoAlert}>
+            {t('No algorithm is available')}
+          </Alert>
+        </Main>
+      );
+    }
+
+    return (
+      <Main>
+        <Grid container justify="center" className={classes.content}>
+          <Grid container justify="center" alignItems="center">
+            <Grid item className={classes.selectLabel}>
+              <Typography>{t('Dataset')}</Typography>
```

Comment:
Instead of this, within `FormControl`, use `InputLabel` as shown in the example here https://material-ui.com/components/selects/#select

Reply:
Or alternatively `FormHelperText`, also in that example